src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -205,11 +204,27 @@ def _create_task_list(
     program_threads: number of threads to use per task
     For instance, if "threads" is set to 128 and "program_threads" to 32, then each task will run 4 batches in parallel.
 
-    We have chosen to use "taskfarmer" for parallelization, which means that "threads" and "program_threads" should
-    have the same value. This ensures that parallelization only happens between tasks, and not within them.
+    For tools capable of processing a batch of genomes, such as GTDB-TK and checkm2,
+    we have chosen to utilize the Slurm THREADS variable within Taskfarmer batch script (submit_taskfarmer.sl) for tool
+    parallelization.
+    From here on we will call this value 'tasks' rather than 'threads' to avoid confusion with operating system
+    threads. It determines the number of simultaneous tool executions per node.
+    'program_threads' is configured to match the CPU count per task execution. To ensure each task handles only one
+    tool execution at a time, both 'threads' and 'program_threads' should be set to the same value, due to the fact the
+    task generator program utilizes 'threads' divided 'program_threads' to determine the quantity of parallel execution
+    in a task.
+    Setting 'threads' and 'program_threads' the same same number ensures that parallelization of tool executions
+    only happens between tasks, and not within tasks.
+    Each tool execution is specified to use 'program_threads' operating system threads.
+
+    For tools that can only handle genomes individually, such as microtrait, mash and eggNOG,
+    we still use the THREADS variable for task parallelization. However, within each task, we further parallelize tool
+    execution using the 'threads' variable. For instance, with 'threads' set to 16 and 'tasks_per_node' set to 4,
+    each node will concurrently execute 4 tasks, with each task executing 16 parallel tool operations.
+    'program_threads' is also utilized to match the operating system threads per tool execution.
```

Comment:
So how is this different from the batch case? We're still using THREADS (tasks) to determine how many simultaneous copies of the tool run, and threads and program threads need to be set to the same value so that only one tool instance runs per THREAD, right? Then the number of of threads each task uses is set by threads == program_threads.

Reply:
in batch case, total num of execution parallelization is `(threads/program_threads) * THREADS ` , we want `<threads/program_threads>` to be 1. So that THREAD controls the num of parallelization. But if threads = 4, program_threads = 2 and THREAD = 3, for each task, there will be 2 (4/2) parallelization of execution and each execution takes 2 cpus. And in total there are 3 tasks in parallel. 

in comparison, in single cases, total num of execution parallelization is `threads * THREADS `.  In each task, there will be threads number of tool execution. And the node run THREADS number of tasks in parallel. 

For both cases, each execution uses program_threads number of OS threads to run.

Reply:
As another example, in eggNOG case, we want to run 16 execution in parallel. I temporarily set THREADS = 1 and threads=16. But setting THREADS = 4 and threads=4 will give us the same performance.

Reply:
Is there any reason we shouldn't, in the future, make the single and batch work the same way and get rid of _threads?

Reply:
yea. Not at all. I think that's the next thing I will be working.

Reply:
Make another stab and making this more clear

Reply:
:+1: